scripts/copy_bundles.py

```diff
@@ -90,24 +92,31 @@ def _parse_args(cls, argv):
         args = parser.parse_args(argv)
         return args
 
-    num_workers = 32
+    num_workers = 8
 
     def __init__(self, argv) -> None:
         super().__init__(num_workers=self.num_workers)
         self.args = self._parse_args(argv)
         self.source = MiniDSS(dss_endpoint=urlunparse(self.args.source),
                               config=Config(max_pool_connections=self.num_workers))
-        self.destination = self._new_dss_client()
-
-    def _new_dss_client(self):
-        return config.dss_client(dss_endpoint=urlunparse(self.args.destination),
-                                 adapter_args=dict(pool_maxsize=self.num_workers))
+        self._destination = None
+        self._destination_expiration = None
+        self._destination_lock = RLock()
+
+    @property
+    def destination(self):
+        with self._destination_lock:
+            if self._destination is None or self._destination_expiration < time.time():
+                self._destination = config.dss_client(dss_endpoint=urlunparse(self.args.destination),
```

Comment:
Let's add a INFO log message here, sth like "Allocating new DSS client for <dss_endpoint> to expire in … seconds."